File: tests/test3.js
// template file
const tc = require("./tc3.json")

// insert Solution class here
var lengthOfLongestSubstring = function(s) {
    var tmp = {},
        currentMaxRange = 0,
        lastRepIndex = 0,
        len = s.length,
        i;
    for(i = 0; i < len; i += 1) {
        var currentChar = s[i];
        if(typeof tmp[currentChar] !== 'undefined') {
            currentMaxRange = Math.max(currentMaxRange, i - lastRepIndex);
            lastRepIndex = Math.max(tmp[currentChar], lastRepIndex);
        } 
        tmp[currentChar] = i + 1;
        
    }
    return Math.max(currentMaxRange, i - lastRepIndex);
};


let isOk = true
const arrInput = tc.input
const arrExpected = tc.expected
for (let i = 0; i < arrInput.length; i++) {
    const solution = lengthOfLongestSubstring(arrInput[i])
    if (solution != arrExpected[i]) {
        isOk = false
        break
    }

}

if (isOk) {
    const output = {
        "result": "OK"
    }
    const data = JSON.stringify(output)
    console.log(data)
}

console.log("test completed")

File: tests/test3.py
# template file
import json
from typing import List


# insert Solution class here
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if len(s) == 0:
            return 0
        res = 1
        slow = 0
        for i in range(1, len(s)):
            if s[i] in s[slow : i]:
                slow = s[slow:i].index(s[i]) + slow + 1
                #print(slow)
            res = max(res, i - slow + 1)
        return res

with open("tc3.json", "r") as read_file:
    data = json.load(read_file)

sol = Solution()
isOk = True

listInput = data["input"]
listExpected = data["expected"]

for i, element in enumerate(listInput):
    solution = sol.lengthOfLongestSubstring(element)
    if solution != listExpected[i]:
        isOk = False
        break

if isOk:
    output = {
        "result": "OK"
    }
    json_object = json.dumps(output, indent=4)
    print(json_object)

print("test completed")